crates/clarirs_num/src/bitvec.rs

```diff
@@ -127,9 +127,17 @@ impl BitVec {
     }
 
     pub fn reverse(&self) -> Self {
-        let mut new_bv = self.words.clone();
-        new_bv.reverse();
-        BitVec::new(new_bv, self.length)
+        let mut new_words = SmallVec::<[u64; 1]>::with_capacity(self.words.len());
+
+        for &word in &self.words {
+            // Reverse the bits within each word
+            let reversed_word = word.reverse_bits();
```

Comment:
So actually this might have been implemented incorrectly before. Reverse should actually reverse bytes, not bits. So let's make sure that is the case here